src/Analyzer/JoinNode.cpp

```diff
@@ -113,4 +118,17 @@ ASTPtr JoinNode::toASTImpl() const
     return tables_in_select_query_ast;
 }
 
+void JoinNode::crossToInner(const QueryTreeNodePtr & join_expression_)
+{
+    if (kind != JoinKind::Cross && kind != JoinKind::Comma)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot rewrite join {} to inner join, expected cross", toString(kind));
+
+    if (children[join_expression_child_index])
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Join expression is not empty: '{}'", children[join_expression_child_index]->formatConvertedASTForErrorMessage());
```

Comment:
Not sure if this need to be `LOGICAL_ERROR`, maybe just `UNSUPPORTED_METHOD`.
Anyway need to add comment to this function in header.

Reply:
I left it as `LOGICAL_ERROR` because for `CROSS` join it's expected to have `nullptr` here, added to comment for function: `Expects the current kind to be CROSS (and join expression to be null because of that)`